client/src/components/admin/tables/ProjectRow.tsx

```diff
@@ -4,16 +4,23 @@ import DeletePopup from './DeletePopup';
 import EditProjectPopup from './EditProjectPopup';
 import useAdminStore from '../../../store';
 import { postRequest } from '../../../api';
+import { getRequest } from '../../../api';
+import FlagsPopup from '../FlagsPopup';
 
 interface ProjectRowProps {
     project: Project;
     idx: number;
+    flags: Flag[];
     checked: boolean;
     handleCheckedChange: (e: React.ChangeEvent<HTMLInputElement>, idx: number) => void;
 }
 
-const ProjectRow = ({ project, idx, checked, handleCheckedChange }: ProjectRowProps) => {
+const ProjectRow = ({ project, idx, flags, checked, handleCheckedChange }: ProjectRowProps) => {
     const [popup, setPopup] = useState(false);
+    const [showFlags, setShowFlags] = useState(false);
+    const [matchingFlagCount, setMatchingFlagCount] = useState(0);
+    const [matchingFlag, setMatchingFlag] = useState<Flag | undefined>(undefined);
+    const [projectIDState, setProjectIDState] = useState('');
```

Comment:
Prob want to name this `flagPopupProjectId` or smth; this is pretty unclear by name what this does.